tests/nn_test.py

```diff
@@ -68,8 +70,8 @@ def testDotProductAttentionInfer(self, dtype, use_bias, causal_mode, impl):
     B, S, T, N, H = 2, 128, 128, 4, 32
     keys = random.split(random.PRNGKey(0), 4)
     Q = random.normal(keys[0], (B, T, N, H), dtype)
-    K = random.normal(keys[1], (B, S, N, H), dtype)
-    V = random.normal(keys[2], (B, S, N, H), dtype)
+    K = random.normal(keys[1], (B, S, N // group_num, H), dtype)
```

Comment:
Can we add a test for GQA/MQA correctness where we just replicate the head_dim of key/value and use standard MHA? This also makes it clear exactly what this is doing.

Reply:
Done. PTAL.